Spanner/src/Operation.php

```diff
@@ -263,9 +281,13 @@ public function executeUpdate(
         $sql,
         array $options = []
     ) {
-        $res = $this->execute($session, $sql, [
-            'transactionId' => $transaction->id()
-        ] + $options);
+        if (!is_null($transaction->id())) {
+            $options += ['transactionId' => $transaction->id()];
+        }
```

Comment:
Why did this logic change?

previously: 
 - if `transactionId` is set in `$options`, it will be overwritten by `$transaction->id()`
 - if `$transaction->id()` is null, it will set `transactionId` in `$options` as null

now:
 - if `transactionId` is set in `$options`, it will overwrite `$transaction->id()`.
 - If `$transaction->id()` is null, `$options['transactionId']` will not be set to `null`

This is a significant change in behavior, and it doesn't seem like the one we want. And if it _is_ the one we want, is the `!is_null` check really necessary?

Reply:
The `executeUpdate` function uses `executeStremingSql` request and here is the request documentation: https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#executesqlrequest

It says the following about transaction selector options:
```
Standard DML statements require a read-write transaction. To protect against replays, single-use transactions are not supported. The caller must either supply an existing transaction ID or begin a new transaction.
```

So, calling this operation with a `null` transaction ID is not right.

The other part of the answer is, the `transactionId` set in `$options` as well as `$transaction->id()` need to be always same. So, precedence may not matter here. I think why this block is added here is, In the odd event of getting a call to this function without any transaction options, it is just making sure that it will go with the transaction id available in the `$transaction` passed.

Why we have the `!is_null` check is, if `begin` is set in the `$options` and the transaction has not yet created, we don't want to send `null` `transactionId` and `begin` in the same request. To avoid confusion, I will change the condition to the following:
```
if (!isset($options['begin'])) {
    $options += ['transactionId' => $transaction->id()];
}
```

or to avoid the change from previous logic, I will change it to the following [**SELECTED**]:
```
if (!isset($options['begin'])) {
    $options = ['transactionId' => $transaction->id()] + $options;
}
```

Reply:
Done!